### PR TITLE
refactor: stx and sip10 send form

### LINKED ISSUE #3268: Reevaluate send form structure
Now that we've rebuilt the send form, and pushed some changes quite quickly, it'd be a nice exercise to go through how we can improve the current structure. Assigned to you and I @fbwoolf, but @edu-stx @alter-eggo you're welcome to participate too.

As it is, there's quite some repetition of logic between forms. All of them, but especially stx/sip10. Now that they've been built, it'll be easy to identify what's duplicated and can be abstracted out & shared. Also, going through stx/sip10 and asking exactly differs between them (a few things, but too much), and creating abstractions there too.

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/4541607132).

Remove code duplications among sip10 and stx send forms

### CODE REVIEW

File: src/app/pages/send/send-crypto-asset-form/form/stacks/use-stacks-common-send-form.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import { stxRecipientValidator } from '@app/common/validation/forms/recipient-validators';
+import { nonceValidator } from '@app/common/validation/nonce-validators';
+import { useNextNonce } from '@app/query/stacks/nonce/account-nonces.hooks';
+import { useCurrentAccountStxAddressState } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+
+import { useSendFormRouteState } from '../../hooks/use-send-form-route-state';
+import { createDefaultInitialFormValues } from '../../send-form.utils';
+
+export function useStacksCommonSendForm(symbol: string, availableTokenBalance: Money) {
```
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/form/stacks/use-stacks-common-send-form.tsx]: Can we use an object here? Think that'll be more readable

File: src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx [hunk 2]
```diff
@@ -1,39 +1,23 @@
...
-import { MemoField } from '../../components/memo-field';
 import { SelectedAssetField } from '../../components/selected-asset-field';
-import { SendCryptoAssetFormLayout } from '../../components/send-crypto-asset-form.layout';
 import { SendMaxButton } from '../../components/send-max-button';
-import { StacksRecipientField } from '../../family/stacks/components/stacks-recipient-field';
-import { defaultSendFormFormikProps } from '../../send-form.utils';
+import { StacksCommonSendForm } from '../stacks/stacks-common-send-form';
 import { useSip10SendForm } from './use-sip10-send-form';
 
 export function StacksSip10FungibleTokenSendForm({}) {
```
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx]:
```suggestion
export function StacksSip10FungibleTokenSendForm() {
```

File: src/app/pages/send/send-crypto-asset-form/form/stacks/stacks-common-send-form.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { StacksRecipientField } from '../../family/stacks/components/stacks-recipient-field';
+import { defaultSendFormFormikProps } from '../../send-form.utils';
+
+interface StacksCommonSendFormProps {
+  onSubmit(
+    values: StacksSendFormValues,
+    formikHelpers: FormikHelpers<StacksSendFormValues>
+  ): Promise<void>;
+  initialValues: StacksSendFormValues;
+  validationSchema: OptionalObjectSchema<{}>;
```
Comment (alter-eggo) [inline src/app/pages/send/send-crypto-asset-form/form/stacks/stacks-common-send-form.tsx:31]: not sure about type here

File: src/app/pages/send/send-crypto-asset-form/components/send-crypto-asset-form.layout.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ export function SendCryptoAssetFormLayout({ children }: SendCryptoAssetFormLayou
   return (
     <Flex
       alignItems="center"
-      data-testId={SendCryptoAssetSelectors.SendForm}
+      data-testid={SendCryptoAssetSelectors.SendForm}
```
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/components/send-crypto-asset-form.layout.tsx:11]: Thx for fixing.

File: src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx (hunk 2 above)
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/form/stacks-sip10/stacks-sip10-fungible-token-send-form.tsx]: Prob can remove `{}` here?

**Review by fbwoolf — APPROVED** [review]
Really nice refactor, @alter-eggo! Thx for the help. 😄

**Comment by fbwoolf** [comment]
Wondering if we should create another issue to think on how the `ordinal-inscription` send form is outside of `send-crypto-asset-form` ...an ordinal is a crypto asset ...so something needs to change there with structure or naming?

File: src/app/pages/send/send-crypto-asset-form/form/stx/stx-send-form.tsx [hunk 5]
```diff
@@ -1,88 +1,47 @@
...
     sendMaxBalance,
-    stxFees,
+    stxFees: fees,
     validationSchema,
   } = useStxSendForm();
 
+  const amountField = (
+    <AmountField
+      balance={availableStxBalance}
+      switchableAmount={<SendFiatValue marketData={stxMarketData} assetSymbol={'STX'} />}
```
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/form/stx/stx-send-form.tsx]:
```suggestion
      switchableAmount={<SendFiatValue marketData={stxMarketData} assetSymbol="STX" />}
```
Would be great if there's a lint rule to enforce string being a plain string

**Review by kyranjamie — APPROVED** [review]

File: src/app/pages/send/send-crypto-asset-form/form/stx/stx-send-form.tsx (hunk 5 above)
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/form/stx/stx-send-form.tsx]: hm, couldn't find rule for this

**Pull request merged**